lib/pull.ml

```diff
@@ -5,17 +5,31 @@ let pull ?(trim_clone = false) ~global_state ~duniverse_dir src_dep =
   let open Duniverse.Repo in
   let { dir; url; hashes; _ } = src_dep in
   let output_dir = Fpath.(duniverse_dir / dir) in
-  let url = Url.to_opam_url url in
-  let open OpamProcess.Job.Op in
-  Opam.pull_tree ~url ~hashes ~dir:output_dir global_state @@| fun result ->
-  let* () = result in
-  if trim_clone then
-    let* () =
-      Bos.OS.Dir.delete ~must_exist:false ~recurse:true
-        Fpath.(output_dir / ".git")
+  if
+    Fpath.(is_prefix (normalize duniverse_dir) (normalize output_dir))
```

Comment:
Maybe make this a separate function `is_in_duniverse_dir` or similar, to explain why it is doing this check through naming.